python_modules/libraries/dagster-dg/dagster_dg/cli/dev.py

```diff
@@ -0,0 +1,201 @@
+import psutil
+import yaml
+
+from dagster_dg.cli.global_options import dg_global_options
+from dagster_dg.config import normalize_cli_config
+from dagster_dg.context import DgContext
+from dagster_dg.error import DgError
+from dagster_dg.utils import DgClickCommand, exit_with_error, pushd
+
+T = TypeVar("T")
+
+_CHECK_SUBPROCESS_INTERVAL = 5
+
+
+@click.command(name="dev", cls=DgClickCommand)
+@click.option(
+    "--code-server-log-level",
+    help="Set the log level for code servers spun up by dagster services.",
+    show_default=True,
+    default="warning",
+    type=click.Choice(["critical", "error", "warning", "info", "debug"], case_sensitive=False),
+)
+@click.option(
+    "--log-level",
+    help="Set the log level for dagster services.",
+    show_default=True,
+    default="info",
+    type=click.Choice(["critical", "error", "warning", "info", "debug"], case_sensitive=False),
+)
+@click.option(
+    "--log-format",
+    type=click.Choice(["colored", "json", "rich"], case_sensitive=False),
+    show_default=True,
+    required=False,
+    default="colored",
+    help="Format of the logs for dagster services",
+)
+@click.option(
+    "--port",
+    "-p",
+    type=int,
+    help="Port to use for the Dagster webserver.",
+    required=False,
+)
+@click.option(
+    "--host",
+    "-h",
+    type=str,
+    help="Host to use for the Dagster webserver.",
+    required=False,
+)
+@click.option(
+    "--live-data-poll-rate",
+    help="Rate at which the dagster UI polls for updated asset data (in milliseconds)",
+    type=int,
+    default=2000,
+    show_default=True,
+    required=False,
+)
+@dg_global_options
+@click.pass_context
+def dev_command(
+    context: click.Context,
+    code_server_log_level: str,
+    log_level: str,
+    log_format: str,
+    port: Optional[int],
+    host: Optional[str],
+    live_data_poll_rate: int,
+    **global_options: Mapping[str, object],
+) -> None:
+    """Start a local deployment of your Dagster project.
+
+    If run inside a deployment directory, this command will launch all code locations in the
+    deployment. If launched inside a code location directory, it will launch only that code
+    location.
+    """
+    cli_config = normalize_cli_config(global_options, context)
+    dg_context = DgContext.from_config_file_discovery_and_cli_config(Path.cwd(), cli_config)
+
+    forward_options = [
+        *_format_forwarded_option("--code-server-log-level", code_server_log_level),
+        *_format_forwarded_option("--log-level", log_level),
+        *_format_forwarded_option("--log-format", log_format),
+        *_format_forwarded_option("--port", port),
+        *_format_forwarded_option("--host", host),
+        *_format_forwarded_option("--live-data-poll-rate", live_data_poll_rate),
+    ]
+
+    if dg_context.is_code_location:
+        cmd = ["uv", "run", "dagster", "dev", *forward_options]
+        temp_workspace_file_cm = nullcontext()
+    elif dg_context.is_deployment:
+        cmd = [
```

Comment:
Can you add comment of "In a deployment context, dg dev will construct a temporary workspace file that points at all defined code locations and uv tool run --with dagster-webserver dagster dev" here and also explain with the `--with` is necessary?